daemon/daemon.go

```diff
@@ -407,20 +407,12 @@ func (daemon *Daemon) reserveName(id, name string) (string, error) {
 
 		conflictingContainer, err := daemon.GetByName(name)
 		if err != nil {
-			if strings.Contains(err.Error(), "Could not find entity") {
-				return "", err
-			}
```

Comment:
The difference is that the user can reuse the name without breaking the containers from other storage driver.
Another possible solution is keep the current behavior but improve the error message to tell the user exactly what had happened and how to delete the bad containers.